llvm/lib/Analysis/DomConditionCache.cpp

```diff
@@ -27,7 +27,8 @@ static void findAffectedValues(Value *Cond,
 
       // Peek through unary operators to find the source of the condition.
       Value *Op;
-      if (match(I, m_PtrToInt(m_Value(Op)))) {
+      if (match(I,
+                m_CombineOr(m_PtrToInt(m_Value(Op)), m_Trunc(m_Value(Op))))) {
```

Comment:
We should also support for pattern for assumes.

Reply:
I'm working on a patch to unify the two findAffectedValues implementations.

Reply:
Rebased on the top of https://github.com/llvm/llvm-project/pull/83161.

Reply:
Can you please also add a test using assume? I think to make it work you need to move the logic from computeKnownBitsFromCond into computeKnownBitsFromCmp.

Reply:
Done.